lib/Service/DocumentService.php

```diff
@@ -83,11 +82,10 @@ class DocumentService {
 	private IRootFolder $rootFolder;
 	private ICache $cache;
 	private IAppData $appData;
-	private ILockingProvider $lockingProvider;
 	private ILockManager $lockManager;
 	private IUserMountCache $userMountCache;
 
-	public function __construct(DocumentMapper $documentMapper, StepMapper $stepMapper, SessionMapper $sessionMapper, IAppData $appData, ?string $userId, IRootFolder $rootFolder, ICacheFactory $cacheFactory, LoggerInterface $logger, ShareManager $shareManager, IRequest $request, IManager $directManager, ILockingProvider $lockingProvider, ILockManager $lockManager, IUserMountCache $userMountCache) {
+	public function __construct(DocumentMapper $documentMapper, StepMapper $stepMapper, SessionMapper $sessionMapper, IAppData $appData, ?string $userId, IRootFolder $rootFolder, ICacheFactory $cacheFactory, LoggerInterface $logger, ShareManager $shareManager, IRequest $request, IManager $directManager, ILockManager $lockManager, IUserMountCache $userMountCache) {
```

Comment:
If touching anyways maybe we can use constructor property promotion directly? 

https://stitcher.io/blog/constructor-promotion-in-php-8